lighty-core/lighty-common/src/main/java/io/lighty/core/common/models/ModuleId.java

```diff
@@ -22,21 +22,33 @@
  */
 public final class ModuleId {
 
+    // Use to describe inside JSON why is required this instance. e.g. "usedBy": "CONTROLLER"
+    private final String usedBy;
     private final XMLNamespace nameSpace;
     private final String name;
     private final Revision revision;
 
     @JsonCreator
-    public ModuleId(@JsonProperty("nameSpace") final String nameSpace, @JsonProperty("name") final String name,
-            @JsonProperty("revision") final String revision) {
+    public ModuleId(@JsonProperty("usedBy") final String usedBy, @JsonProperty("nameSpace") final String nameSpace,
+            @JsonProperty("name") final String name, @JsonProperty("revision") final String revision) {
+        this(usedBy, nameSpace, name, Revision.ofNullable(revision).orElse(null));
+    }
+
+    public ModuleId(@JsonProperty("nameSpace") final String nameSpace,
```

Comment:
Done

Reply:
There is allowed only one `@JsonCreator` annotation per class. I have removed @JsonProperty annotation from this constructor, since is used only for testing.